lib/recording/RecordingManager.js

```diff
@@ -266,54 +264,48 @@ class RecordingManager {
-            let deferred = Q.defer();
-            return backendClient.getEntryInfo(entryId)
-                .then((entryObj)=> {
-
-                    if (entryObj.recordedEntryId) {
-                        recordedEntryId = entryObj.recordedEntryId;
-                        logger.debug("[%s] Found recordedEntryId : %s", entryId, recordedEntryId)
-                        return Q.resolve(recordedEntryId)
-                    }
-                    else {
-                        return Q.reject("recordedEntryId not exist on entry object")
-                    }
-                })
-                .then((recordedEntryId)=> {
-                    logger.debug("[%s-%s][replaceUnknownRecordedEntry][2 of 4] About to call changeRecordedEntry on recordingEntrySession", entryId, recordedEntryId);
-
-                    recordingEntrySession.serializedActionsPromise = deferred.promise; //lock the ability to change playlistJson, in order to avoid race-condition
-                    return recordingEntrySession.changeRecordedEntry(recordedEntryId);
-                }).then(()=> {
-                    logger.debug("[%s-%s][replaceUnknownRecordedEntry][3 of 4] About to change directory name from [%s] to [%s]", entryId, recordedEntryId, oldRecordingSessionPath, recordingEntrySession.recordingSessionPath)
-                    return qio.rename(oldRecordingSessionPath, recordingEntrySession.recordingSessionPath)
-                })
-                .then(()=> {
-                    logger.debug("[%s-%s][replaceUnknownRecordedEntry][4 of 4] About to export it to live directory", entryId, recordedEntryId);
-                    return recordingEntrySession.accessibleRecording()
-                })
-                .catch((err)=> {
-                    logger.error("[%s] Failed to changeRecordedEntry: %s", entryId, ErrorUtils.error2string(err))
-                })
-                .finally(()=> {
-                    logger.info("[%s-%s][replaceUnknownRecordedEntry] Finished successfully", entryId, recordedEntryId);
-                    recordingEntrySession.changeRecordingIsRunning = false;
-                    deferred.resolve()
-
-                })
-        }
-        catch (err){
-            logger.error("Failed to changeRecordedEntry: %s", ErrorUtils.error2string(err))
+        if (recordingEntrySession.unknownReplacementInProcess === true){
+            return Q.reject(`[${recordingEntrySession.entryId}] Replacement of UNKNOWN entry is already running`);
         }
+        recordingEntrySession.unknownReplacementInProcess = true;
+        logger.debug(`[${recordingEntrySession.entryId}] [replaceUnknownRecordedEntry][1 of 4] About to call get entry`);
+        let entryId = recordingEntrySession.entryId;
+        let oldRecordingSessionPath = recordingEntrySession.recordingSessionPath;
+        let recordedEntryId;
+        let deferred = Q.defer();
+        return backendClient.getEntryInfo(entryId)
+            .then((entryObj) => {
```

Comment:
should test if the folder exists before changing the recordedEntry + should not spin here